workflow/util/util.go

```diff
@@ -244,7 +246,21 @@ func SubmitWorkflow(wfIf v1alpha1.WorkflowInterface, wf *wfv1.Workflow, opts *Su
 	if err != nil {
 		return nil, err
 	}
-	return wfIf.Create(wf)
+	if opts.DryRun {
+		// Keep the workflow metadata because it will be removed by the Post request
+		workflowTypeMeta := wf.TypeMeta
+		err := wfClientset.ArgoprojV1alpha1().RESTClient().Post().
+			Namespace(wf.Namespace).
+			Resource("workflows").
+			Body(wf).
+			Param("dryRun", "All").
```

Comment:
For k8s version < 1.12, without feature gate enable.  will this post the real action to the apiServer ?

Should we limit this action on k8s version < 1.12

Reply:
I tried on a few different versions of kubernetes on minikube and this is what I got:
- On v1.10.13:  It does not complain and creates the resources ignoring completely the --dry-run option
- On v1.12.10: ```Failed to submit workflow: the dryRun alpha feature is disabled```
- On >= v1.13: It works as expected

It seems that a check should be added when using --dry-run with an kubernetes version older than 1.12